js/widgets/forms/clearButton.js

```diff
@@ -20,59 +25,56 @@ define( [
 		_create: function() {
 			this._super();
 
+			if ( !this.inputNeedsWrap ) {
+				return;
+			}
```

Comment:
Actually, we don't want to short-circuit if it's not a search input. Adding a clear button to a plain textinput is perfectly valid.